packages/rpc-methods/src/restricted/manageAccounts.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  SubjectType,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+  PermissionSpecificationBuilder,
+} from '@metamask/permission-controller';
+import { Json, NonEmptyArray } from '@metamask/utils';
+import { ethErrors } from 'eth-rpc-errors';
+
+export const methodName = 'snap_manageAccounts';
+
+export type ManageAccountsMethodHooks = {
+  /**
+   * Gets the snap keyring implementation.
+   */
+  getSnapKeyring: (snapOrigin: string) => Promise<{
+    handleKeyringSnapMessage: (
+      snapId: string,
+      message: any,
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      saveSnapKeyring: Function,
+    ) => Promise<Json>;
+  }>;
+
+  /**
+   * Saves the snap keyring, should be called after mutable operations.
+   *
+   * If an account was removed the address of the account must be passed to
+   * update the UI.
+   */
+  saveSnapKeyring: (removedAddress?: string) => Promise<void>;
```

Comment:
I don't think this type is up to date.

Reply:
Updated in [bbc422d](https://github.com/MetaMask/snaps/pull/1290/commits/bbc422da04d6d9a9cf16b927bd99042a69a5e354)